keyrings/codeartifact.py

```diff
@@ -156,13 +156,7 @@ def get_password(self, service, username):
             name=repository_name,
         )
 
-        # Create session with any supplied configuration.
-        session = boto3.Session(
-            region_name=region,
-            profile_name=config.get("profile_name"),
-            aws_access_key_id=config.get("aws_access_key_id"),
-            aws_secret_access_key=config.get("aws_secret_access_key"),
-        )
+        session = self.get_boto_session(region=region, config=config)
```

Comment:
As part of 537fe6f, I've created a [`_get_codeartifact_client`](https://github.com/jmkeyes/keyrings.codeartifact/commit/537fe6fcd6d928383cdfe9b0f7453acbb1b27054#diff-b826bb69a2cf12b8184eb81cca663aa40aac73a97692f0b34112fbd0963edd89R172) instance method that returns an initialized CodeArtifact service client with specific options.